app/main.py

```diff
@@ -1,2 +1,65 @@
+from typing import Any
+
+
+class Node:
+    def __init__(self, key: Any, value: Any) -> None:
+        self.key = key
+        self.value = value
+        self.next = None
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length: int = 0
+        self.capacity: int = 8
+        self.hash_table: list[Node | None] = [None] * self.capacity
```

Comment:
The use of `Node | None` in the type hint for `hash_table` requires Python 3.10 or later. If you need compatibility with earlier versions, consider using `Optional[Node]` from the `typing` module instead.